verify/go.mod

```diff
@@ -1,22 +1,169 @@
 module github.com/containers/podman-machine-os
 
-go 1.23.0
+go 1.23.3
+
+toolchain go1.24.3
 
 require (
+	github.com/containers/podman/v5 v5.5.0
```

Comment:
I would strongly prefer if we do not depend on podman.

We should just parse `podman machine info` once to get the provider then, first we safe a bunch of vendor code and needless renovate bumps but more importantly the vendored podman code will be a different version compared to the actual binary is use of the provider detection logic changes it will break

Reply:
Sure, I will do that. Of course, this would be simpler if this code lived in c/podman. We should not need to reinvent the wheel.

Reply:
Done.